lib/services/template-service.ts

```diff
@@ -83,97 +82,89 @@ export class TemplateService implements ITemplateService {
     }
 
     getTemplates(): Promise<Array<any>> {
-        // let tempDetails: Array<any> = [];
-        // const promises: Array<any> = [];
+        let tempDetails: Array<any> = [];
+        const promises: Array<any> = [];
 
         return new Promise((resolve, reject) => {
-            resolve(Backup.fallback);
-            // tmpCache.get("tempDetails", (error: any, value: any) => {
-            //     if (!error) {
-            //         if (value === undefined) {
-            //             this.getTemplatesNames()
-            //                 .then((templateNames: any) => {
-            //                     return templateNames;
-            //                 })
-            //                 .then((names: any) => {
-            //                     names.forEach((name: string) => {
-            //                         promises.push(
-            //                             this.getAppTemplateDetails(name)
-            //                                 .then((details) => {
-            //                                     tempDetails.push(details);
-            //                                 })
-            //                                 .catch((errorDetails) => {
-            //                                     reject(errorDetails);
-            //                                 })
-            //                         );
-            //                     });
-
-            //                     Promise.all(promises)
-            //                         .then(() => {
-            //                             tempDetails = this.sortTmpData(tempDetails);
-            //                             tmpCache.set("tempDetails", tempDetails, Config.cacheTime);
-            //                             resolve(tempDetails);
-
-            //                         })
-            //                         .catch((errorPromises: any) => {
-            //                             // TODO Implement error logger
-            //                             resolve(Backup.fallback);
-            //                         });
-            //                 })
-            //                 .catch((errorTemplates: any) => {
-            //                     console.error(errorTemplates);
-            //                 });
-            //         } else {
-            //             // Load data from cache
-            //             resolve(value);
-            //         }
-            //     }
-            // });
+            tmpCache.get("tempDetails", (error: any, value: any) => {
+                if (!error) {
```

Comment:
I think we should consider refactoring / restructuring / reordering this method as its indentation is on its way to the pyramid of doom.

Reply:
I will think of a way to refactor this method, however, all those checks are necessary in order to provide the template data from Cache, Github, or fallback. Maybe we could think of something when we refactor the Caching logic in a separate PR?